Delete Torch::deploy from pytorch

PaliC opened the pull request:
Summary: Removes torch::deploy from pytorch core. The new home of torch::deploy can be found at [https://github.com/pytorch/multipy](https://github.com/pytorch/multipy).

Test Plan: This shouldn't break anything hopefully. Currently all internal users are migrated to multipy/runtime.

seemethere reviewed (approved):
Approving for github workflow changes / build system changes

PaliC committed:
Delete Torch::deploy from pytorch (#85443)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/85443

Removes torch::deploy from pytorch core

Test Plan:
This shouldn't break anything hopefully. Currently all internal users are migrated to multipy/runtime. Also
cant_test_if_deleted

Reviewed By: seemethere

Differential Revision: D39710545

fbshipit-source-id: b85344d6598a4888897d5bf9f0a1b5b75dd44421

malfet reviewed (commented):
Please consider removing reference to torch_deploy from lines 25-27 of __init__.py

d4l3k commented:
@malfet we still need the code in __init__.py -- please do not delete

d4l3k commented on .github/workflows/pull.yml:
```diff
@@ -302,26 +302,6 @@ jobs:
       docker-image-name: pytorch-linux-focal-py3.7-gcc7
       build-generates-artifacts: false
 
-  linux-bionic-cuda11_6-py3_10-gcc7-deploy-build:
```
do you think internal tests are enough or do we want to add a sanity check to build multipy from torch to help avoid regressions?

PaliC replied:
Personally I would, but after talking to @seemethere, didn't opt to as it would make the pytorch devX worse by adding more test time. Though if the test is lightweight, it may be reasonably to add a sanity check for this reason.

malfet commented:
@PaliC you need to land it using internal workflows. Also, looks like PR is out of sync with trunk
